### DISCUSSION TITLE
Shuffle MultipleChoice

### DISCUSSION THREAD

**Discussion by ThreeCO** [memoetapp/memoet#51] [Ideas]
I am wondering whether you do not shuffle the options of multiple choice questions on purpose.
I have never used elixir yet. But as far as I understand the concept in order to shuffle it should be sufficient to add.

` note.options = Enum.shuffle note.options `
in memoet/lib/memoet_web/controllers/note_controller.ex

most probably in this part

@spec show(Plug.Conn.t(), map) :: Plug.Conn.t()
  def show(conn, %{"id" => id, "deck_id" => deck_id}) do
    user = Pow.Plug.current_user(conn)
    deck = Decks.get_deck!(deck_id)
    note = Notes.get_note!(id)
    _++++ SHOULD BE ADDED HERE?! ++++_
    **note.options = Enum.shuffle note.options**
    _++++ SHOULD BE ADDED HERE?! ++++_
    if note.user_id != user.id do

Am I right with that assumption? If so, would you accept that change?

**Comment by manhtai** [comment]
Yes it would be easy to implement. But I have to think about adding a configuration for enable / disable shuffle feature for notes because some decks have questions that come in order. Like this:

A. Option 1
B. Option 2
C. 2 options above

If we shuffle all quetions then "2 options above" will have a chance to be in the first and will cause confusion.

**Reply by ThreeCO** [reply]
Good point. So it would probably be a "shuffle-able" checkbox with is or is not enabled in its default setting.
I would assume it would make sense to **usually enable shuffling** and **explicitly disable it if it's not needed**.

`Is it OK to shuffle the Options? (*) Yes / ( ) No`

**Reply by manhtai** [reply]
Maybe it should be a deck-level option. It will affect all notes in that deck. Would you be alright with that option?

**Reply by ThreeCO** [reply]
Sure. It would be easier to implement and also easier to maintain on a per deck level basis.
If I went down the "not shuffle" road, I would know that I could use questions like "2 options above".

Of course implementation on a note-level would mean more flexibility. But I assume it's not needed.

**Reply by manhtai** [reply]
I’ve created an issue here #52 for implement, feel free to pick it if you want to learn some Elixir, it’s sure fun!

**Reply by ThreeCO** [reply]
I'm coming from php, js and a little python.
I seem to not get how Elixir is working at all :-/

At least I've already learned, that it's not gonna be as easy as i thought.
I keep getting errors when trying to shuffle the (assumed) list in the place I mentioned in my original post.

Gonna try a little further ... but I assume you should not hope for a solution / PR coming from me :-D

**Comment by manhtai** [comment]
I've implemeted the feature on #54. Please try it out!

**Reply by ThreeCO** [reply]
Thank you. Works like a charm!
And it indeed was way more complex than I thought 😅
